### PR TITLE
Refactor Person to StudySpot, AddressBook to StudyTracker

### PR DESCRIPTION
**Pull request by qreoct**
Refactor more instances of `Person` to `StudySpot`, and `AddressBook` to `StudyTracker`. Part of #11. 

Also made changes to:
- Changed default data file to `data/studytracker.json` from `data/addressbook.json`.
- Main source code, including comments + Test files.
- Partial changes to User Guide and Developer Guide terminology.

### CODE REVIEW

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
     public static final String MESSAGE_UNKNOWN_COMMAND = "Unknown command";
     public static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format! \n%1$s";
-    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The person index provided is invalid";
-    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d persons listed!";
+    public static final String MESSAGE_INVALID_PERSON_DISPLAYED_INDEX = "The study spot index provided is invalid";
```
Comment (joenzkimchan) [inline src/main/java/seedu/address/commons/core/Messages.java:10]: Should be named MESSAGE_INVALID_STUDYSPOT_DISPLAYED_INDEX

File: src/main/java/seedu/address/model/studyspot/Name.java [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import static seedu.address.commons.util.AppUtil.checkArgument;
 
 /**
- * Represents a Person's name in the address book.
+ * Represents a Person's name in the study tracker.
```
Comment (joenzkimchan) [inline src/main/java/seedu/address/model/studyspot/Name.java:7]: Should represent a studyspot's name

File: src/main/java/seedu/address/storage/JsonSerializableStudyTracker.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import seedu.address.model.StudyTracker;
+import seedu.address.model.studyspot.StudySpot;
+
+/**
+ * An Immutable StudyTracker that is serializable to JSON format.
+ */
+@JsonRootName(value = "studytracker")
+class JsonSerializableStudyTracker {
+
+    public static final String MESSAGE_DUPLICATE_PERSON = "StudySpots list contains duplicate study spot(s).";
```
Comment (joenzkimchan) [inline src/main/java/seedu/address/storage/JsonSerializableStudyTracker.java:22]: should be MESSAGE_DUPLICATE_STUDYSPOT

File: src/main/java/seedu/address/logic/commands/EditCommand.java [hunk 4]
```diff
@@ -50,16 +50,16 @@
             + PREFIX_PHONE + "91234567 "
             + PREFIX_EMAIL + "johndoe@example.com";
 
-    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited StudySpot: %1$s";
+    public static final String MESSAGE_EDIT_PERSON_SUCCESS = "Edited study spot: %1$s";
```
Comment (joenzkimchan) [inline src/main/java/seedu/address/logic/commands/EditCommand.java:53]: MESSAGE_EDIT_STUDYSPOT_SUCCESS

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 5]
```diff
@@ -33,17 +33,17 @@
             + PREFIX_TAG + "friends "
             + PREFIX_TAG + "owesMoney";
 
-    public static final String MESSAGE_SUCCESS = "New person added: %1$s";
-    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+    public static final String MESSAGE_SUCCESS = "New study spot added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This study spot already exists in the study tracker";
```
Comment (joenzkimchan) [inline src/main/java/seedu/address/logic/commands/AddCommand.java:37]: MESSAGE_DUPLICATE_STUDYSPOT

File: src/main/java/seedu/address/commons/core/Messages.java [hunk 6]
```diff
@@ -7,7 +7,7 @@
...
+    public static final String MESSAGE_PERSONS_LISTED_OVERVIEW = "%1$d study spot(s) listed!";
```
Comment (joenzkimchan) [inline src/main/java/seedu/address/commons/core/Messages.java:11]: MESSAGE_STUDYSPOTS_LISTED_OVERVIEW

**Review by joenzkimchan — CHANGES_REQUESTED** [review]
All tests pass, just some small changes

**Review by mslevis — COMMENTED** [review]
Good job! just missed out some variables

**Review by mslevis — APPROVED** [review]

**Review by joenzkimchan — APPROVED** [review]
LGTM!

**Pull request merged**